### ISSUE TITLE
Kontalk does not receive images from another client xmpp (Gajim)

### ISSUE THREAD

**Issue by susobaco** [kontalk/androidclient#1053]
### Expected behavior
Kontalk should show the preview of the image

### Actual behavior
Does not show anything

### Steps to reproduce
1. I upload an image from gajim using "http upload"
2. Kontalk does not show anything
3.

### Environment

Kontalk version: F-droid 4.0.4

Android version: 4.4.2

Device model: Primux Omega 6

### Logs

Attach Captures

![pantallazo](https://user-images.githubusercontent.com/5635692/29740164-4b0e994e-8a50-11e7-907e-7c97e7b6bb7f.png)

![img_20170826_110946127](https://user-images.githubusercontent.com/5635692/29740167-518a0506-8a50-11e7-9d69-a280ebf59f4a.jpeg)

**Comment by daniele-athome** [comment]
Thanks for reporting. This might have to do with how Gajim sends images. I just took a quick read to the [Gajim plugin](https://dev.gajim.org/gajim/gajim-plugins/blob/master/httpupload/httpupload.py). It seems like it is sending the correct message info for the originale image itself, but the thumbnail is placed inside a XHTML block, which is unsupported by Kontalk.
If you long-click on the message, do you have the download action in the toolbar? That should appear at least, allowing you to actually download the image.

**Comment by susobaco** [comment]
No, there is simply no message in kontalk. The http link does not appear.

**Comment by daniele-athome** [comment]
Oh, I thought that in the screenshot was the URL to the image. I'll do some debug with Gajim myself then :-)

**Comment by daniele-athome** [comment]
Doing some debug with Gajim I discovered the following:

* Gajim sends the URL in a oob packet, so it's ok to download for Kontalk (and it works, but only for https URLs)
* The preview image is sent inside a XHTML body which is unsupported by Kontalk (we use [XEP-0231](https://xmpp.org/extensions/xep-0231.html), we think it's a better approach in this case)

The URL is not displayed in Kontalk, just "Image: [type]", which is correct. You can download the image if you click on the message.
As for the thumbnail, I'm not sure supporting XHTML is the best way to go (especially for security).

Please use the latest beta (4.1.0-beta1) to do some more tests and report back so we can see this issue through (I'll reopen the issue). Thank you!

**Comment by daniele-athome** [comment]
In the meantime, I've enabled non-HTTPS download (that is, plain HTTP) through #1071.

**Comment by susobaco** [comment]
I have done the same test with kontalk beta and I have obtained the same result, the image is not previewed, nor can you see the link, the rest of the messages, they work perfectly.

**Comment by daniele-athome** [comment]
Ok then, could you send me the debug log generated by Kontalk when receiving the image please? Maybe there is something wrong with the Gajim version we are using.

**Comment by susobaco** [comment]
Ok, Gajim 0.16.8 
[kontalk-android.log](https://github.com/kontalk/androidclient/files/1330269/kontalk-android.log)

**Comment by daniele-athome** [comment]
There was a problem with the logging system so the log you produced is not helping. I'll publish beta2 within the next few days (probably tonight or tomorrow actually) with the logging fix so we can have a better log. Please send one as soon as you see the beta published. Thanks for your patience!

**Comment by susobaco** [comment]
No problem! I'll set the log as soon as I get the f-droid version. A pleasure to collaborate in the project, and thank you for your work!

**Comment by susobaco** [comment]
I
[kontalk-android.log](https://github.com/kontalk/androidclient/files/1362645/kontalk-android.log)


n kontalk 4.1.0.-beta 3 still without preview the image and without receiving the link, attached log

**Comment by daniele-athome** [comment]
The message with the image is coming in, but it's not processed by the app:

```xml
<message xmlns="jabber:client" xml:lang="ca" id="199" type="chat" to="[your-jid]" from="[sender-jid]">
  <request xmlns="urn:xmpp:receipts"/>
  <x xmlns="jabber:x:oob">
    <url>https://suchat.org:5443/[...]/terminal.jpg</url>
  </x>
  <body>https://suchat.org:5443/[...]/terminal.jpg</body>
  <thread>cfCZvuOjpRIhpmopPEftstWTHvhqKsXj</thread>
</message>
```

It has probably something to do with the `<x/>` element, I'll do some tests.

**Comment by daniele-athome** [comment]
Weirdly enough, it was this part the caused the bug:

```
id="199"
```

The ID is used to generate an internal filename for the image. It extracts the first 5 characters and it goes out of bounds (because the id is 3 characters long), but that doesn't get notified by Smack somehow so the problem wasn't clear.

I have no workaround at the moment but changing client with another that uses message IDs longer than 4 characters (Pidgin should do), but the fix is already here and will be released with the next beta (or probably 4.1.0 final). Thanks for your help and patience!!!
